pjsip/src/pjsua-lib/pjsua_acc.c

```diff
@@ -4269,6 +4269,31 @@ pj_status_t pjsua_acc_handle_call_on_ip_change(pjsua_acc *acc)
 
                 status = pjsua_call_hangup(i, PJSIP_SC_GONE, NULL, NULL);
 
+                if (pjsua_var.ua_cfg.cb.on_ip_change_progress) {
+                    pjsua_ip_change_op_info info;
+
+                    pj_bzero(&info, sizeof(info));
+                    info.acc_hangup_calls.acc_id = acc->index;
+                    info.acc_hangup_calls.call_id = call_info.id;
+
+                    pjsua_var.ua_cfg.cb.on_ip_change_progress(
+                                                             acc->ip_change_op,
+                                                             status,
+                                                             &info);
+                }
+            }
+            else if (call_info.role==PJSIP_ROLE_UAC &&
+                     call_info.state < PJSIP_INV_STATE_CONFIRMED)
+            {
```

Comment:
why not combine it with the above?

Reply:
- For readibility, because they are conceptually different. The above is intentional hangup via configured settings, the latter is forced hangup due to specific conditions.
- Also the actions can also be different, for example with the current patch, `pjsua_call_hangup()` which will send CANCEL and CANCEL may not be useful if remote does not receive the INVITE or remote have sent its final answer (either success/2xx or failure/3xx-6xx), so perhaps terminating the call forcefully somehow may be better.